src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,39 @@
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+            return country;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Unable to add county " + country, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            Country movie = session.get(Country.class, id);
```

Comment:
The variable `movie` should be renamed to `country` to accurately reflect the entity being retrieved. This will improve code readability and maintain consistency with the method's purpose.